programs/psa/CMakeLists.txt

```diff
@@ -1,12 +1,15 @@
-add_executable(crypto_examples crypto_examples.c)
-target_link_libraries(crypto_examples mbedtls)
+set(executables
+    crypto_examples
+    key_ladder_demo
+    psa_constant_names
+)
 
-add_executable(key_ladder_demo key_ladder_demo.c)
-target_link_libraries(key_ladder_demo mbedtls)
+foreach(exe IN LISTS executables)
+    add_executable(${exe} ${exe}.c)
+    target_link_libraries(${exe} mbedtls)
```

Comment:
The dependency on mbedtls came from 4fa9f9f744f3e4f1c1d9045e952f2c42d9da32d3 which was a bulk change from linking with mbedcrypto to linking with all the libraries. At the moment, all programs in `programs/psa` are crypto only, so it would make sense to link them with libmbedcrypto. That they are linked with libmbedtls is arguably a defect in https://github.com/ARMmbed/mbedtls/pull/3085.

Reply:
I've changed that in a dedicated commit.